### ISSUE TITLE
Authenticator componentDidMount() doesn't call this.checkUser()

### ISSUE THREAD

**Issue by D0berm4n** [aws-amplify/amplify-js#3195]
**Describe the bug**
[Authenticator](https://github.com/aws-amplify/amplify-js/blob/master/packages/aws-amplify-react/src/Auth/Authenticator.jsx) `componentDidMount` method doesn,t call `this.checkUser()` in case if local storage contains `Constants.SIGN_IN_WITH_HOSTEDUI_KEY` because `localStorage.getItem` method returns string "false"

**To Reproduce**
1. login
2. refresh page
3. see nothing

**Screenshots**
![Image of Yaktocat](http://i64.tinypic.com/a40qs0.png)

**Desktop (please complete the following information):**
 - OS: ubuntu
 - Browser chrome
 - Version 74.0.3729.108

**Comment by manueliglesias** [comment]
Hi @D0berm4n 

Thank you for the bug report, looks like this check should be something similar to `byHostedUI === 'false'`

**Comment by MTibby92** [comment]
Getting the same behavior as well

**Comment by jordanranz** [comment]
This is fixed in the latest unstable version of `aws-amplify-react`:
https://github.com/aws-amplify/amplify-js/commit/cc75ea176e05e8c3e483317f98f8f299599117d9

We will publish this to our npm stable version early next week.

**Comment by icrustandi** [comment]
Hi - I'm still experiencing the same symptoms with the same steps to reproduce. It seems like after the first initial login, the 'amplify-signin-with-hostedUI' key in localstorage still exists/set to 'true', and on the refresh after login, the amplify-signin-with-hostedUI key still exists, and this.checkUser isn't called.

If I remove this key manually after authState==='signedIn', then the refresh works as expected. My understanding is after the cognito redirect has happened, the amplify-signin-with-hostedUI should have been removed right after so that future refreshes work normally. This doesn't seem to be happening

**Comment by jukkahuuskonen** [comment]
Same problem seems to exist.
As far as I've looked into it the problem seems to be in Authenticator componentDidMount() function:

```
componentDidMount() {
        const config = this.props.amplifyConfig;
        if (config) {
            Amplify.configure(config);
        }
        this._isMounted = true;
        // the workaround for Cognito Hosted UI
        // don't check the user immediately if redirected back from Hosted UI
        // instead waiting for the hub event sent from Auth module
        // the item in the localStorage is a mark to indicate whether
        // the app is redirected back from Hosted UI or not
        const byHostedUI = localStorage.getItem(Constants.SIGNING_IN_WITH_HOSTEDUI_KEY);
        localStorage.removeItem(Constants.SIGNING_IN_WITH_HOSTEDUI_KEY);
        if (byHostedUI !== 'true') this.checkUser();
    }
```

If i've understood correctly this sets the byHostedUI (TRUE) first and then removes it from local storage. Then immediately after checks the user (this.checkUser) IF byHosted !== true.

So, if the byHosted was set to TRUE, the user will never be checked since the checks happens only in componentDidMount which runs only once, hence with hostedUi it never gets checked. Unless the app unmounts and remounts the Authenticator-component.

There should be a check implemented in componentDidUpdate(), that checks if the Constants.SIGNING_IN_WITH_HOSTEDUI_KEY has been removed from the localStorage and then runs  this.checkUser(). That should fix the login problem. This would require storing the byHostedUI-variable in state and check for changes on it.

**Comment by haverchuck** [comment]
@D0berm4n You're using the HostedUI in conjunction with the react Authenticator component?  Is this to leverate the 'Greetings/Signout' component or for something else?

**Comment by mivassenko** [comment]
@haverchuck I don't use hostedUI

**Comment by jordanranz** [comment]
I believe this will be fixed by: https://github.com/aws-amplify/amplify-js/pull/4258

**Comment by jordanranz** [comment]
This should be fixed with the PR listed above. Please open a new issue if you have any related questions or concerns.

**Comment by sadrataban** [comment]
> Hi - I'm still experiencing the same symptoms with the same steps to reproduce. It seems like after the first initial login, the 'amplify-signin-with-hostedUI' key in localstorage still exists/set to 'true', and on the refresh after login, the amplify-signin-with-hostedUI key still exists, and this.checkUser isn't called.
> 
> If I remove this key manually after authState==='signedIn', then the refresh works as expected. My understanding is after the cognito redirect has happened, the amplify-signin-with-hostedUI key should have been removed right after so that future refreshes work normally. This doesn't seem to be happening.
> 
> I'm using Auth.federatedSignIn({provider: 'Google'}) to sign users in, if that makes any difference

I tried to do exactly what you did. But the issue still exists. Could you please share your code snippet?

**Comment by mitchconquer** [comment]
> > Hi - I'm still experiencing the same symptoms with the same steps to reproduce. It seems like after the first initial login, the 'amplify-signin-with-hostedUI' key in localstorage still exists/set to 'true', and on the refresh after login, the amplify-signin-with-hostedUI key still exists, and this.checkUser isn't called.
> > If I remove this key manually after authState==='signedIn', then the refresh works as expected. My understanding is after the cognito redirect has happened, the amplify-signin-with-hostedUI key should have been removed right after so that future refreshes work normally. This doesn't seem to be happening.
> > I'm using Auth.federatedSignIn({provider: 'Google'}) to sign users in, if that makes any difference
> 
> I tried to do exactly what you did. But the issue still exists. Could you please share your code snippet?

I'm also still having the same problem as described here.
